gobblin-aws/src/main/java/gobblin/aws/AWSJobConfigurationManager.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.aws;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Maps;
+import com.google.common.eventbus.EventBus;
+import com.typesafe.config.Config;
+
+import gobblin.annotation.Alpha;
+import gobblin.cluster.GobblinClusterConfigurationKeys;
+import gobblin.cluster.GobblinHelixJobScheduler;
+import gobblin.cluster.JobConfigurationManager;
+import gobblin.cluster.event.NewJobConfigArrivalEvent;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.ExecutorsUtils;
+import gobblin.util.SchedulerUtils;
+import static gobblin.aws.GobblinAWSUtils.*;
+
+
+/**
+ * Class for managing AWS Gobblin job configurations.
+ *
+ * <p>
+ *   This class reads all the job configuration at startup from S3
+ *   and schedules a refresh to poll from S3 for any new job configurations.
+ *   The jobs read are scheduled by the {@link GobblinHelixJobScheduler} by posting a
+ *   {@link NewJobConfigArrivalEvent} for each job configuration file.
+ * </p>
+ *
+ * @author Abhishek Tiwari
+ */
+@Alpha
+public class AWSJobConfigurationManager extends JobConfigurationManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AWSJobConfigurationManager.class);
+
+  private Optional<String> jobConfS3Uri;
+  private Map<String, Properties> jobConfFiles;
+
+  private final long refreshIntervalInSeconds;
+
+  private final ScheduledExecutorService fetchJobConfExecutor;
+
+  public AWSJobConfigurationManager(EventBus eventBus, Config config) {
+    super(eventBus, config);
+    this.jobConfFiles = Maps.newHashMap();
+    if (config.hasPath(GobblinAWSConfigurationKeys.JOB_CONF_REFRESH_INTERVAL_IN_SECONDS)) {
```

Comment:
Suggestion: support refresh interval as a time value config and support notation like: 30s, 20m .